Formula/gdal.rb

```diff
@@ -84,7 +83,7 @@ class Gdal < Formula
     depends_on "swig" => :build
   end
 
-  option "without-python", "Build without python2 support"
+  depends_on "libpng"
```

Comment:
Probably, especially since there are cases where a dependency has to be declared before another.

For instance,
```
depends_on "foo" => :optional
depends_on "bar" if build.with? "foo"
```
would break if `depends_on "foo" => :optional` is moved below `depends_on "bar" if build.with? "foo"` because the `with-foo` option doesn't exist until it's declared by `depends_on "foo" => :optional`

Also, we generally keep `=> :build` dependencies at the top of the list, then required, then recommended, and then optional.